### PR TITLE
added tests for UserService

### PR DESCRIPTION
**Pull request by RamiSh**
Getting my head wrapped around the spring framework. Adding tests for the `UserService` service.

### CODE REVIEW

**Comment by RamiSh** [comment]
tests for the `save` methods will be added in a separate pr.

File: api/src/test/java/com/qis/api/service/UserServiceTests.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+        Assertions.assertEquals(userAsJson, retrievedUserAsJson);
+    }
+
+    @Test
+    void findById_WhenUserDoesNotExist_ReturnsNull() {
+        // Arrange
+        Mockito
+                .when(userRepository.findById(Mockito.anyString()))
+                .thenReturn(null);
```
Comment (aarnaout) [inline api/src/test/java/com/qis/api/service/UserServiceTests.java:61]: The actual repository will return an empty optional not null
```suggestion
                .thenReturn(Optional.empty());
```

File: api/src/test/java/com/qis/api/service/UserServiceTests.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+        // Act
+        Optional<User> retrievedUser = userService.findById("test_user_id");
+
+        // Assert
+        Assertions.assertNull(retrievedUser);
```
Comment (aarnaout) [inline api/src/test/java/com/qis/api/service/UserServiceTests.java:67]:
```suggestion
        Assertions.assertFalse(retrievedUser.isPresent());
```

File: api/src/test/java/com/qis/api/helpers/UserHelper.java [hunk 3]
```diff
@@ -8,7 +8,7 @@
 public class UserHelper {
 	public static final String userTestAllFields1Id = "1316d445-d7fc-474f-8747-2e0a7323e99a";
 
-	public static User getTestAllFields1() {
+	public static User createTestUser() {
```
Comment (aarnaout) [inline api/src/test/java/com/qis/api/helpers/UserHelper.java:11]: This method is used in multiple places, can you do a refactor to update all of them?

Comment (RamiSh) [reply api/src/test/java/com/qis/api/helpers/UserHelper.java:11]: There are two other methods with the same name defined in different classes [here](https://github.com/aarnaout/quran-in-salat/blob/734160ebfff2a7984c20cb8bb08b171c74c827a1/api/src/test/java/com/qis/api/helpers/GlobalCounterHelper.java#L10) and [here](https://github.com/aarnaout/quran-in-salat/blob/734160ebfff2a7984c20cb8bb08b171c74c827a1/api/src/test/java/com/qis/api/helpers/EntryCounterHelper.java#L11). This one doesn't have references outside of the `UserServiceTests`

Comment (RamiSh) [reply api/src/test/java/com/qis/api/helpers/UserHelper.java:11]: Closing

Comment (RamiSh) [reply api/src/test/java/com/qis/api/helpers/UserHelper.java:11]: As for the rest, they can by renamed when we touch their respective helpers/tests.

**Pull request merged**